### PR TITLE
Fix/shard static sync

### PR DESCRIPTION
**Pull request by JungleKim**
- Add `get_origin_data` and `set_revision` for inject sync logic.
- Fix Testcase

### CODE REVIEW

File: shard_static/models.py [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-    last_modified = models.DateTimeField(null=False, verbose_name='Last Modified', )
 
+class BaseShardStaticModel(ShardStaticMixin, models.Model):
     objects = ShardStaticManager()
 
     class Meta:
         abstract = True
+
+    @classmethod
+    def get_origin_data(cls, status):
```
Comment (antiline) [inline shard_static/models.py:28]: This function is supposed to be in the manager, not in the model.

Comment (JungleKim) [reply shard_static/models.py:28]: If this function will be in the manager, the manager is depend on the status model.
I don't want what the manager is depend on the status model.

I thought that this function gets revision value and calls manager function like `cls.objects.find_by_last_modified()`.

Also I thought that this function is allow to call directly queryset method like `cls.objects.filter()`.

Comment (antiline) [reply shard_static/models.py:28]: I don't want what the manager is depend on the status model. <- why?

If you do not want to put it in the manager, how about creating a related service?

Comment (antiline) [inline shard_static/models.py:28]: Need return type. How about `List`?

Comment (antiline) [reply shard_static/models.py:28]: `data` is plural.
But this function name seems to return a single instance.
How about `get_origins` or `find_origin_data`, `filter_origin_data`?

Comment (antiline) [reply shard_static/models.py:28]: And also need `status` type or change variable name. it is very formal name .

File: shard_static/models.py [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
-    diffusible = False
 
+    shard_group = ALL_SHARD_GROUP
     objects = ShardStaticManager()
 
     class Meta:
-        db_table = 'static_sync_status'
+        abstract = True
 
+    def set_revision(self, shard_static_model):
```
Comment (antiline) [inline shard_static/models.py]: BaseStaticSyncStatus -> shard_static_model -> BaseShardStaticModel -> status .....
Is there better way?

Comment (antiline) [reply shard_static/models.py]: how about `record_status` for function name.

File: shard_static/services/sync_static_service.py [hunk 3]
```diff
@@ -39,18 +39,15 @@ def sync_static(model_name: str, database_alias: str):
...
-    if origin_data.count() >= config.SHARD_SYNC_MAX_ITEMS:
+    if origin_data.count() >= config.SHARD_STATIC_SYNC_MAX_ITEMS:
         raise TooManySyncItemsException()
 
     with transaction.atomic(database_alias):
-        last_modified = None
         for _data in origin_data:
-            if not last_modified or last_modified < _data.last_modified:
-                last_modified = _data.last_modified
+            sync_status.set_revision(_data)
```
Comment (antiline) [inline shard_static/services/sync_static_service.py]: It need `order by` or check what is bigger.

File: shard_static/models.py (hunk 2 above)
Comment (antiline) [reply shard_static/models.py]: need command about check new value

**Pull request closed without merging**